### DISCUSSION TITLE
DetailsList - dynamically add and remove columns

### DISCUSSION THREAD

**Discussion by LukasSchauffele** [microsoft/powercat-creator-kit#305] [Q&A] [answered]
Hi all

Is it possible to dynamically add and remove columns to and from the DetailsList?

I would like to hide/remove columns on small screens. I'm thinking something like `ColVisible` which can be dynamically set like so

```
Parent.Size >= ScreenSize.Medium
```

**Comment by MizVik** [comment]
You can do that, or at least I have my plan to do the same thing.
You have to do that logic in the columns parameter:

If(Parent.Size >= ScreenSize.Medium;
  Table(...); //column definition for large screen
  Table(...) //column definistion for small screen
)

**Comment by LukasSchauffele** [comment, marked as answer]
Thanks for the suggestion @MizVik.

I figured out that by setting the `ColVisible` to a blank string the column is excluded and lets me show and hide columns dynamically.

```yaml
# Setting ColName to an empty string hides the column
ColName: If(
            'Home Screen'.Size >= ScreenSize.Medium,
            "InspectionDateTime",
            ""
        )
```

I think it would be neater to have an explicit column property called `ColVisible`. I'll create an issue for an enhancement.

**Comment by LukasSchauffele** [comment]
Created feature request in below issue:

[https://github.com/microsoft/powercat-creator-kit/issues/307](https://github.com/microsoft/powercat-creator-kit/issues/307)
